Recursive subsyntax causes detectindent to hang

```lua
-- mod-version:3
local syntax = require "core.syntax"

syntax.add {
  name = "Hang",
  files = { "%.hang$" },
  patterns = {
    { pattern = { "{", "}" }, type = "<T>", syntax = ".hang" },
  },
  symbols = {},
}
```
Using this syntax file, and opening a corresponding file will cause the editor to hang before giving the following error, if `<T>` is not `comment` or `string`.
```
[ERROR] /usr/local/share/lite-xl/plugins/detectindent.lua:242: /usr/local/share/lite-xl/core/common.lua:231: stack overflow at /usr/local/share/lite-xl/core/init.lua:1222

stack traceback:
[C]: in for iterator 'for iterator'
/usr/local/share/lite-xl/plugins/detectindent.lua:242: in upvalue 'detect_indent_stat'
/usr/local/share/lite-xl/plugins/detectindent.lua:264: in upvalue 'update_cache'
/usr/local/share/lite-xl/plugins/detectindent.lua:279: in function 'core.doc.new'
/usr/local/share/lite-xl/core/object.lua:54: in function 'core.titleview.__call'
/usr/local/share/lite-xl/core/init.lua:1143: in function 'core.open_doc'
/usr/local/share/lite-xl/plugins/treeview.lua:614: in function </usr/local/share/lite-xl/plugins/treeview.lua:609>
[C]: in function 'xpcall'
/usr/local/share/lite-xl/core/init.lua:1221: in function 'core.try'
/usr/local/share/lite-xl/plugins/treeview.lua:609: in field 'perform'
...(skipping 8 levels)
/usr/local/share/lite-xl/core/keymap.lua:205: in function 'core.keymap.on_key_pressed'
/usr/local/share/lite-xl/core/keymap.lua:246: in function 'core.keymap.on_mouse_pressed'
/usr/local/share/lite-xl/core/init.lua:1246: in upvalue 'on_event'
/usr/local/share/lite-xl/plugins/macro.lua:19: in function 'core.on_event'
[C]: in function 'xpcall'
/usr/local/share/lite-xl/core/init.lua:1221: in function 'core.try'
/usr/local/share/lite-xl/core/init.lua:1306: in function 'core.step'
/usr/local/share/lite-xl/core/init.lua:1384: in function 'core.run'
(...tail calls...)
[string "local core..."]:9: in function <[string "local core..."]:2>
[C]: in function 'xpcall'
[string "local core..."]:2: in main chunk
```

--- a/data/plugins/detectindent.lua
+++ b/data/plugins/detectindent.lua
@@ -75,7 +75,9 @@ local function escape_comment_tokens(token)
 end
 
 
-local function get_comment_patterns(syntax)
+local function get_comment_patterns(syntax, _loop)
+  _loop = _loop or 1
+  if _loop > 5 then return end
   if comments_cache[syntax] then
     if #comments_cache[syntax] > 0 then
       return comments_cache[syntax]
@@ -125,7 +127,7 @@ local function get_comment_patterns(syntax)
     elseif pattern.syntax then
       local subsyntax = type(pattern.syntax) == 'table' and pattern.syntax
         or core_syntax.get("file"..pattern.syntax, "")
-      local sub_comments = get_comment_patterns(subsyntax)
+      local sub_comments = get_comment_patterns(subsyntax, _loop + 1)
       if sub_comments then
         for s=1, #sub_comments do
           table.insert(comments, sub_comments[s])

